skrub/_join_aggregator.py

```diff
@@ -0,0 +1,354 @@
+        elif op_name in CATEG_OPS:
+            categ_ops.append(op_name)
+        else:
+            ValueError(
+                f"'ops' options are {ALL_OPS}, got: {op_name}."
+            )
+    return num_ops, categ_ops
+
+
+class AssemblingEngine:
+    """Helper class to perform the join and aggregate operations.
+
+    This is an abstract base class that is specialized depending
+    on the module of the dataframe used (Pandas or Polars).
+
+    This helper class is in charge of mapping strings to aggregation
+    operators of the respective module.
+    """
+    @classmethod
+    def get_for(cls, tables):
+        """Returns the AssemblingEngine implementation for given tables
+        module.
+
+        Parameters
+        ----------
+        tables : List of Tuple of (table, cols_to_join, cols_to_agg)
+            We only use table to detect the module used.
+        
+        Returns
+        -------
+        assembling_engine : AssemblingEngine
+            The suited AssemblingEngine implementation.
+        """
+        use_pandas = all(
+            [isinstance(table, pd.DataFrame) for table, _, _ in tables]
+        )
+        use_polars = False
+        if POLARS_SETUP:
+            # we don't mix DataFrame and LazyFrame
+            use_polars = (
+                all(
+                    [isinstance(table, pl.DataFrame) for table, _, _ in tables]
+                )
+                or all(
+                    [isinstance(table, pl.LazyFrame) for table, _, _ in tables]
+                )
+            )
+        if use_pandas:
+            return PandasAssemblingEngine()
+        elif use_polars:
+            return PolarsAssemblingEngine()
+        else:
+            raise NotImplementedError(
+                "Only Pandas or Polars DataFrame are currently supported."
+            )
+
+    @abstractmethod
+    def agg(self, table, cols_to_join, cols_to_agg, ops):
+        pass
+
+    @abstractmethod
+    def join(self, left, right, left_on, right_on):
+        pass
+
+
+def pandas_split_num_categ_cols(table):
+    num_cols = table.select_dtypes("number").columns
+    categ_cols = table.select_dtypes(
+        ["object", "string", "category"]
+    ).columns
+    return num_cols, categ_cols
+    
+
+def pandas_get_agg_ops(cols, agg_ops):
+    pandas_ops_mapping = {
+        "mode": pd.Series.mode
+    }
+    stats = {}
+    for col, op_name in product(cols, agg_ops):
+        op = pandas_ops_mapping.get(op_name, op_name)
+        stats[f"{col}_{op_name}"] = pd.NamedAgg(col, op)
+    return stats
+
+
+class PandasAssemblingEngine(AssemblingEngine):
+
+    def agg(self, table, cols_to_join, cols_to_agg, agg_ops):
+        
+        num_cols, categ_cols = pandas_split_num_categ_cols(
+            table[cols_to_agg]
+        )
+        num_ops, categ_ops = split_num_categ_ops(agg_ops)
+
+        num_stats = pandas_get_agg_ops(num_cols, num_ops)
+        categ_stats = pandas_get_agg_ops(categ_cols, categ_ops)
```

Comment:
I'm surprise that the above work, as I see the same code applied for categorical columns and numerical ones, but the dictionary above seemed to map only to categorical operations.

Reply:
Yes, the beauty of `pd.NamedAgg(col, op)` is being able to use `"mean"`, `np.mean` or even `pd.Series.mode` for `op` :)
Therefore, we can use most `agg_ops` as strings. `"mode"`, however, is a little bit trickier, hence the mapping.